src/lib/store.js

```diff
@@ -324,27 +195,13 @@ export default class Store {
 			} catch (err) {
 				log.error('Failed writing external consent data', err);
 			}
-			// Write vendor cookie to appropriate domain
-			this.mergeVendorConsentsToGlobalCookie();
-			this.mergePurposeConsentsToGlobalCookie();
-
-			const globalVendorConsents = {...globalVendorConsentData, vendorList};
-			writeVendorConsentCookie(globalVendorConsents);
-
-			this.persistedGlobalVendorConsentData = copyData(globalVendorConsentData);
 		} else  {
-			// Write vendor cookie to appropriate domain
-			writeVendorConsentCookie(vendorConsents);
-
-			// Write publisher cookie if enabled
-			if (config.storePublisherData) {
-				writePublisherConsentCookie(publisherConsents);
-			}
+			writeConsentCookie(encodedConsent);
 		}
 
-		// Store the persisted data
-		this.persistedVendorConsentData = copyData(vendorConsentData);
-		this.persistedPublisherConsentData = copyData(publisherConsentData);
+		this.persistedConsentString = encodedConsent;
+		this.persistedConsentData = decodeConsentData(encodedConsent);
+		this.cmpApi.update(encodedConsent);
```

Comment:
Shouldn't this function be called like this?
```suggestion
		this.cmpApi.update(encodedConsent, true);
```
According to documentation of `CmpApi.prorotype.update`:

> `uiVisible` - default false.  set to true if the ui is being shown with this tc string update, this will set the correct values for eventStatus and displayStatus